lib/text-class/src/Data/Text/Class.hs

```diff
@@ -116,6 +118,19 @@ instance FromText Natural where
 instance ToText Natural where
     toText = T.pack . show
 
+instance FromText Word32 where
+    fromText =
+        validate <=< (fmap fromIntegral . fromText @Natural)
```

Comment:
This will not work :scream:! The `fromIntegral` that transform from `Natural` to `Word32` needs to happen _after_ the validation to be safe. Otherwise, this may just throw if the value can't be converted back. 

Note that @hasufell is doing something similar but re-using constructor from the `text` package which makes things a bit nicer: https://github.com/input-output-hk/cardano-wallet/pull/2213/files#diff-b892068f732dfe3e67cd6f7a464975ba21a3588de3e5ea74022be5dfaa18e476

I'd suggest cherry-picking this change :+1: